ocp_resources/user_defined_network.py

```diff
@@ -128,3 +139,88 @@ def to_dict(self) -> None:
 
             if self.ipam_lifecycle:
                 _layer2["ipamLifecycle"] = self.ipam_lifecycle
+
+
+class Layer3Subnets:
+    """
+    UserDefinedNetwork layer3 subnets object.
+
+    API reference:
+    https://ovn-kubernetes.io/api-reference/userdefinednetwork-api-spec/#layer3subnet
+    """
+
+    def __init__(
+        self,
+        cidr: Optional[str] = None,
+        host_subnet: Optional[int] = None,
+    ):
+        """
+        UserDefinedNetwork layer3 subnets object.
+
+        Args:
+            cidr (Optional[str]): CIDR specifies L3Subnet, which is split into smaller subnets for every node.
+            host_subnet (Optional[int]): host_subnet specifies the subnet size for every node.
+        """
+        self.cidr = cidr
+        self.host_subnet = host_subnet
+
+
+class Layer3UserDefinedNetwork(UserDefinedNetwork):
+    """
+    UserDefinedNetwork layer3 object.
+
+    API reference:
+    https://ovn-kubernetes.io/api-reference/userdefinednetwork-api-spec/#layer3config
+    """
+
+    def __init__(
+        self,
+        role: str,
+        mtu: Optional[int] = None,
+        subnets: Optional[List[Layer3Subnets]] = None,
+        join_subnets: Optional[List[str]] = None,
+        **kwargs,
+    ):
+        """
+        Create and manage UserDefinedNetwork with layer3 configuration
+
+        Args:
+            role (Optional[str]): role describes the network role in the pod. Required.
+            mtu (Optional[int]): mtu is the maximum transmission unit for a network.
+            subnets (Optional[List[Layer3Subnets]]): subnets are used for the pod network across the cluster.
+            join_subnets (Optional[List[str]]): join_subnets are used inside the OVN network topology.
+        """
+        super().__init__(
+            topology=TopologyType.LAYER3,
+            **kwargs,
+        )
+        self.role = role
+        self.mtu = mtu
+        self.subnets = subnets
+        self.join_subnets = join_subnets
+
+    def to_dict(self) -> None:
+        super().to_dict()
+        if not self.kind_dict and not self.yaml_file:
+            if not self.role:
+                raise MissingRequiredArgumentError(argument="role")
+
+            self.res["spec"][TopologyType.LAYER3.lower()] = {"role": self.role}
+            _layer3 = self.res["spec"][TopologyType.LAYER3.lower()]
+
+            if self.mtu:
+                _layer3["mtu"] = self.mtu
+
+            if self.join_subnets:
+                _layer3["joinSubnets"] = self.join_subnets
+
+            if self.subnets is not None:
+                _subnets = _layer3.setdefault("subnets", [])
+
+                for subnet in self.subnets:
+                    subnet_dict = {
```

Comment:
I don't understand this one, `self.subnets` is already a list of dicts.

Reply:
It's a list of object class, and I must create from it a valid dictionary.